sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/HiveThriftServer2Listener.scala

```diff
@@ -229,9 +229,9 @@ private[thriftserver] class HiveThriftServer2Listener(
     if (countToDelete <= 0L) {
       return
     }
-    val view = kvstore.view(classOf[ExecutionInfo]).index("finishTime").first(0L)
+    val view = kvstore.view(classOf[ExecutionInfo]).index("closeTime").first(0L)
     val toDelete = KVUtils.viewToSeq(view, countToDelete.toInt) { j =>
-      j.finishTimestamp != 0
+      j.closeTimestamp != 0
```

Comment:
Thanks for catching this bug!

I think the correct check here would be `!isExecutionActive(j)`
with renaming:
```
  def isLiveExecutionActive(execInfo: LiveExecutionData): Boolean = {
    !(execInfo.state == ExecutionState.FAILED ||
      execInfo.state == ExecutionState.CANCELED ||
      execInfo.state == ExecutionState.CLOSED)
  }

  def isExecutionActive(execInfo: ExecutionInfo): Boolean = {
    !(execInfo.state == ExecutionState.FAILED ||
      execInfo.state == ExecutionState.CANCELED ||
      execInfo.state == ExecutionState.CLOSED)
  }
```

We do want to remove executions that are FAILED or CANCELED, even when they don't get CLOSED, because many applications may just don't care and don't close failed/cancelled statements properly. We don't want them to be stuck forever in the UI. Nothing other than closing will ever happen to these executions...
We should also wipe LiveExecutionData this way, and then then in `onOperationClosed` check if `executionList(e.id)` still exists, or ignore it otherwise.

We should not delete executions that are FINISHED, because those can still be fetching results.

Reply:
> Thanks for catching this bug!

Finding this bug is a bit expensive...， we should run 600w sql today .. but only run 100w now...